dask/dataframe/shuffle.py

```diff
@@ -248,8 +248,7 @@ def shuffle(df, index, shuffle=None, npartitions=None, max_branch=32, compute=No
         shuffle=shuffle,
         compute=compute,
     )
-    del df3["_partitions"]
-    return df3
```

Comment:
I would hope that our use of `del` more generally would also benefit from this.  So hopefully this modification wouldn't be necessary.